### PR TITLE
[Enhancement]  Support select star and expr without alias in CreateMaterializedViewStatement

### PR DESCRIPTION
**Pull request by LiShuMing**
Signed-off-by: Kevin Li <ming.moriarty@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

- Support select star and expr without alias in CreateMaterializedViewStatement, like other Database system:
  - https://docs.snowflake.com/en/user-guide/views-materialized.html#label-limitations-on-creating-materialized-views
  - https://www.postgresql.org/docs/current/sql-creatematerializedview.html

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by murphyatwork — COMMENTED** [review]
What if column name changed ?

**Comment by LiShuMing** [comment]
> What if the column name changed?

For now, the column name can not be changed in SR for now. But schema change can add/drop columns of base tables.

If mv's columns contain the changed column, the mv will be set inactive. 

But now the rule is simple, and will not find the expr column in the base table. 

I've added two simple test cases for the cases above.

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewTest.java [hunk 1]
```diff
@@ -1524,27 +1525,165 @@ public void testAsHasStar() {
                 ") " +
                 "as select k1 ss, *  from tbl1;";
         try {
-            UtFrameUtils.parseStmtWithNewParser(sql, connectContext);
+            StatementBase statementBase = UtFrameUtils.parseStmtWithNewParser(sql, connectContext);
```
Comment (ABingHuang) [inline fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewTest.java:1528]: do you has the case to test two table join with the same column name each.
select a.*, b.* from a join b
and a and b both have column col1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 28 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectListItem.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Util.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|19|19|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewTest.java (hunk 1 above)
Comment (LiShuMing) [reply fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewTest.java:1529]: add `testAsHasStarWithSameColumns` and `testMVWithSameColumns`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 28 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectListItem.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Util.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|19|19|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 28 / 29 (96.55%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|[679]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectListItem.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Util.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|19|19|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

**Comment by LiShuMing** [comment]
run starrocks_fe_unittest

**Comment by LiShuMing** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 41 / 42 (97.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|[679]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectListItem.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Util.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|19|19|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SelectAnalyzer.java [hunk 2]
```diff
@@ -241,6 +241,8 @@ private List<Expr> analyzeSelect(SelectList selectList, Relation fromRelation, b
                 }
                 outputFields.addAll(fields);
 
+                item.setStarExpandedExprs(
```
[the remark this answers is not in the document]
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SelectAnalyzer.java]: fixed.

**Review by kangkaisen — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 37 / 38 (97.37%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|[679]|
|:large_blue_circle:|com/starrocks/common/util/Util.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|19|19|100.00%|[]|

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**